src/ObjCommon/Game/IW5/Json/JsonLeaderboardDef.h

```diff
@@ -0,0 +1,99 @@
+                                 {
+                                     {LBCOL_TYPE_NUMBER,    "number"   },
+                                     {LBCOL_TYPE_TIME,      "time"     },
+                                     {LBCOL_TYPE_LEVELXP,   "levelxp"  },
+                                     {LBCOL_TYPE_PRESTIGE,  "prestige" },
+                                     {LBCOL_TYPE_BIGNUMBER, "bignumber"},
+                                     {LBCOL_TYPE_PERCENT,   "percent"  },
+                                     {LBCOL_TYPE_TIME_FULL, "time_full"},
+    });
+
+    NLOHMANN_JSON_SERIALIZE_ENUM(LbAggType,
+                                 {
+                                     {LBAGG_TYPE_MIN,  "normal"},
+                                     {LBAGG_TYPE_MAX,  "rank"  },
+                                     {LBAGG_TYPE_SUM,  "sum"   },
+                                     {LBAGG_TYPE_LAST, "last"  },
+    });
+
+    NLOHMANN_JSON_SERIALIZE_ENUM(LbUpdateType,
+                                 {
+                                     {LBUPDATE_TYPE_NORMAL,  "normal" },
+                                     {LBUPDATE_TYPE_RANK,    "rank"   },
+                                     {LBUPDATE_TYPE_COMBINE, "combine"},
+    });
+
+    class JsonColumnDef
+    {
+    public:
+        std::optional<std::string> name;
+        int id;
+        int propertyId;
+        bool hidden;
+        std::optional<std::string> statName;
+        LbColType type;
+        int precision;
+        LbAggType agg;
+        int uiCalColX;
+        int uiCalColY;
+
+        inline void to_json(nlohmann::json& out, const JsonColumnDef& in)
+        {
+            optional_to_json(out, "name", in.name);
+            out["id"] = in.id;
+            out["propertyId"] = in.propertyId;
+            out["hidden"] = in.hidden;
+            optional_to_json(out, "statName", in.statName);
+            out["type"] = in.type;
+            out["precision"] = in.precision;
+            out["agg"] = in.agg;
+            out["uiCalColX"] = in.uiCalColX;
+            out["uiCalColY"] = in.uiCalColY;
+        }
+
+        inline void from_json(const nlohmann::json& in, JsonColumnDef& out)
+        {
+            optional_from_json(in, "name", out.name);
+            in.at("id").get_to(out.id);
+            in.at("propertyId").get_to(out.propertyId);
+            in.at("hidden").get_to(out.hidden);
+            optional_from_json(in, "statName", out.statName);
+            in.at("type").get_to(out.type);
+            in.at("precision").get_to(out.precision);
+            in.at("agg").get_to(out.agg);
+            in.at("uiCalColX").get_to(out.uiCalColX);
+            in.at("uiCalColY").get_to(out.uiCalColY);
+        };
+    };
+
+    NLOHMANN_DEFINE_TYPE_EXTENSION(JsonColumnDef, name, id, propertyId, hidden, statName, type, agg, uiCalColX, uiCalColY);
+
+    class JsonLeaderboardDef
+    {
+    public:
+        std::optional<std::string> name;
```

Comment:
The name of the asset should already be given by the filename and specified in the `.zone`. Having it in the file again makes the value be specified in two places which makes it annoying and unintuitive to change.